src/ambulance_game/markov/waiting.py

```diff
@@ -401,75 +414,30 @@ class 2 individuals or the overall of both
     )
     pi = get_markov_state_probabilities(pi, all_states, output=np.ndarray)
     if class_type == 3:
-        mean_waiting_time_class_1 = mean_waiting_time_formula(
+        mean_waiting_time = overall_waiting_time_formula(
             all_states=all_states,
             pi=pi,
-            class_type=1,
+            class_type=class_type,
             lambda_2=lambda_2,
             lambda_1=lambda_1,
             mu=mu,
             num_of_servers=num_of_servers,
             threshold=threshold,
             system_capacity=system_capacity,
             buffer_capacity=buffer_capacity,
-            formula=formula,
+            waiting_formula=waiting_formula,
         )
-        mean_waiting_time_class_2 = mean_waiting_time_formula(
+    else:
+        mean_waiting_time = waiting_formula(
             all_states=all_states,
             pi=pi,
-            class_type=2,
+            class_type=class_type,
             lambda_2=lambda_2,
             lambda_1=lambda_1,
             mu=mu,
             num_of_servers=num_of_servers,
             threshold=threshold,
             system_capacity=system_capacity,
```

Comment:
All this can be simplified (because the signatures are the same) to be something like:

```
if condition:
    get_time = waiting_formula
if other_condition:
    get_time = overall_waiting_time_formula
if ...
return get_time(all_states=all_states,
              pi=pi,
              class_type=2,
              class_type=class_type,
              lambda_2=lambda_2,
              lambda_1=lambda_1,
              mu=mu,
              num_of_servers=num_of_servers,
              threshold=threshold,
             system_capacity=system_capacity,)
```

I'm leaving out precision here but hopefully the idea is clear (if not let me know).